cairo/tests/conftest.py

```diff
@@ -266,3 +269,35 @@ def pytest_collection_modifyitems(session, config, items):
             item.add_marker(pytest.mark.skip(reason="Cached results"))
 
     yield
+
+
+def pytest_assertrepr_compare(op, left, right):
+    """
+    Custom assertion comparison for EVM objects to provide detailed field-by-field comparison.
+    """
+    if not (
+        hasattr(left, "__class__")
+        and hasattr(right, "__class__")
+        and left.__class__.__name__ == "Evm"
+        and right.__class__.__name__ == "Evm"
+        and op == "=="
+    ):
+        return None
+
+    lines = []
+    for field in fields(left):
+        left_val = getattr(left, field.name)
+        right_val = getattr(right, field.name)
+
+        if field.name != "error":
+            # Regular field comparison
+            if left_val != right_val:
+                lines.extend(
+                    [
+                        f"{field.name} field mismatch:",
+                        f"  left:  {left_val}",
+                        f"  right: {right_val}",
+                    ]
+                )
```

Comment:
missing error case

and (
                str(self.error) == str(other.error)
                if self.error is not None
                else isinstance(self.error, type(other.error))
            )